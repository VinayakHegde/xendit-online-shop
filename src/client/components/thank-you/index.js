export * from './thank-you-container';
