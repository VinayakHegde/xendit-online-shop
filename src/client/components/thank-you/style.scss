
.ThankYou {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  &__container {
    margin: 100px 0 126px;
    text-align: center;
  }
  &__image {
    display: block;
    height: 100px;
    margin: 16px auto 0;
  }

  &__message {
    margin: 8px 0 24px;
  }

  .button {
    max-width: 236px;
  }

  @media (min-width: 768px) {
    .button {
      max-width: 200px;
    }
  }
}
