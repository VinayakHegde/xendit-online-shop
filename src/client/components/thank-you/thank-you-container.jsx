import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import { clearCart } from "../../store/cart-slice";
import { ThankYou } from "./thank-you";

export const ThankYouContainer = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(clearCart());
  }, []);
  return (
    <ThankYou />
  );
}
