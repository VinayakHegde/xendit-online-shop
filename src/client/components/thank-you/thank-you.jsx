import React from "react";
import { Link } from "react-router-dom";
import { Button } from "../button";
import './style.scss';

export const ThankYou = () => (
  <div className="ThankYou">
    <div className="ThankYou__container">
      <img src={`/images/success.svg`} className="ThankYou__image" alt="success" />
      <img src="/images/thank-you.svg" className="ThankYou__image" alt="thank-you" />
      <p className="ThankYou__message">
        <span>Your order has been successfully placed.</span>
      </p>
      <Link to="/" className="button">
        <Button type="button" title="Return Home" />
      </Link>
    </div>
  </div>
);
