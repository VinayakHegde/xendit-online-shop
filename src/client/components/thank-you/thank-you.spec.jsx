import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { ThankYou } from "./thank-you";

describe('ThankYou', () => {
  it('should render without crash', () => {
    const { container } = render(<BrowserRouter><ThankYou /></BrowserRouter>)
    expect(container).toMatchSnapshot();
  });
});
