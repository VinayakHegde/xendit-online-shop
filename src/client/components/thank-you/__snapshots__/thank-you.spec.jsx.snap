// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYou should render without crash 1`] = `
<div>
  <div
    class="ThankYou"
  >
    <div
      class="ThankYou__container"
    >
      <img
        alt="success"
        class="ThankYou__image"
        src="/images/success.svg"
      />
      <img
        alt="thank-you"
        class="ThankYou__image"
        src="/images/thank-you.svg"
      />
      <p
        class="ThankYou__message"
      >
        <span>
          Your order has been successfully placed.
        </span>
      </p>
      <a
        class="button"
        href="/"
      >
        <button
          class="Button undefined"
          title="Return Home"
          type="button"
        >
          Return Home
        </button>
      </a>
    </div>
  </div>
</div>
`;
